### PR TITLE
Add underline and bold for navigation improvements

### PR DESCRIPTION
**Pull request by Hmikihiro**
#619
I implemented only "(A) highlight current page in navbar"
image

### CODE REVIEW

File: website/src/header.rs [hunk 1]
```diff
@@ -129,10 +129,15 @@ pub fn HamburgerMenu<G: Html>() -> View<G> {
         ) {
             NavLinks {}
             (if let Some((version, data)) = sidebar.get_clone() {
+                let sidebar_current = SidebarCurrent{
+                    version:version.unwrap_or_else(|| "next".to_string()),
+                    path:"".to_owned(),
+                    data,
+                };
```
Comment (lukechu10) [inline website/src/header.rs:136]: Nit: fixed formatting. Unfortunately we don't have automatic formatting inside view! macro yet...

Also changed `"".to_owned()` to `String::new()` for consistency with the rest of the codebase.
```suggestion
                let sidebar_current = SidebarCurrent {
                    version: version.unwrap_or_else(|| "next".to_string()),
                    path: String::new(),
                    data,
                };
```

File: website/src/main.rs [hunk 2]
```diff
@@ -82,17 +84,19 @@ fn switch<G: Html>(route: ReadSignal<Routes>) -> View<G> {
             }
         },
         Routes::Docs(a, b) => {
-            let data = fetch_docs_data(format!("/static/docs/{a}/{b}.json"));
+            let data = fetch_docs_data(format!("/static/docs/{a}/{b}.json",));
```
Comment (lukechu10) [inline website/src/main.rs:87]: Seems like this was added by accident?
```suggestion
            let data = fetch_docs_data(format!("/static/docs/{a}/{b}.json"));
```

File: website/src/main.rs [hunk 3]
```diff
@@ -82,17 +84,19 @@ fn switch<G: Html>(route: ReadSignal<Routes>) -> View<G> {
...
+            let path: Signal<String> = create_signal(format!("{a}/{b}"));
```
Comment (lukechu10) [inline website/src/main.rs:88]: Type inference should work well enough here.
```suggestion
            let path = create_signal(format!("{a}/{b}"));
```

File: website/src/main.rs [hunk 4]
```diff
@@ -82,17 +84,19 @@ fn switch<G: Html>(route: ReadSignal<Routes>) -> View<G> {
...
                                 data=data.clone(),
-                                sidebar=(
-                                    "next".to_string(),
-                                    cached_sidebar_data.1.clone(),
-                                ),
+                                sidebar=SidebarCurrent{
+                                    version : "next".to_string(),
+                                    path : path.get_clone(),
+                                    data:cached_sidebar_data.1.clone(),
+                                },
```
Comment (lukechu10) [inline website/src/main.rs:99]: Nit: formatting
```suggestion
                                sidebar=SidebarCurrent {
                                    version: "next".to_string(),
                                    path: path.get_clone(),
                                    data: cached_sidebar_data.1.clone(),
                                },
```

File: website/src/main.rs [hunk 5]
```diff
@@ -106,17 +110,19 @@ fn switch<G: Html>(route: ReadSignal<Routes>) -> View<G> {
         Routes::VersionedDocs(version, a, b) => {
             let version = version.clone();
             let data = fetch_docs_data(format!("/static/docs/{version}/{a}/{b}.json"));
+            let path: Signal<String> = create_signal(format!("{a}/{b}"));
```
Comment (lukechu10) [inline website/src/main.rs:113]: See previous.
```suggestion
            let path = create_signal(format!("{a}/{b}"));
```

File: website/src/main.rs [hunk 6]
```diff
@@ -106,17 +110,19 @@ fn switch<G: Html>(route: ReadSignal<Routes>) -> View<G> {
...
             view! {
                 (if let Some(data) = data.get_clone() {
                     if let Some(cached_sidebar_data) = cached_sidebar_data.get_clone() {
                         let version = version.clone();
                         view! {
                             content::Content(
                                 data=data.clone(),
-                                sidebar=(
+                                sidebar=SidebarCurrent{
```
Comment (lukechu10) [inline website/src/main.rs:121]:
```suggestion
                                sidebar=SidebarCurrent {
```

File: website/src/main.rs [hunk 7]
```diff
@@ -106,17 +110,19 @@ fn switch<G: Html>(route: ReadSignal<Routes>) -> View<G> {
...
                                     version,
-                                    cached_sidebar_data.1.clone(),
-                                ),
+                                    path:path.get_clone(),
+                                    data:cached_sidebar_data.1.clone(),
```
Comment (lukechu10) [inline website/src/main.rs:124]:
```suggestion
                                    path: path.get_clone(),
                                    data: cached_sidebar_data.1.clone(),
```

File: website/src/sidebar.rs [hunk 8]
```diff
@@ -18,20 +18,29 @@ pub struct SidebarData {
...
-    let sections = data
+pub fn Sidebar<G: Html>(sidebar: SidebarCurrent) -> View<G> {
+    let sections = sidebar.data
         .sections
         .into_iter()
         .map(|SidebarSection { title, items }| {
             let pages = items
                 .into_iter()
                 .map(|SidebarItem { name, href }| {
+                    let selected = if href == sidebar.path{"font-bold underline"}else{""};
```
Comment (lukechu10) [inline website/src/sidebar.rs:37]:
```suggestion
                    let selected = if href == sidebar.path { "font-bold underline" } else { "" };
```

**Review by lukechu10 — COMMENTED** [review]
This is great! Only comments are about formatting code. Otherwise looks good to me. If there's nothing else, I'll apply the suggestions and merge.

**Comment by Hmikihiro** [comment]
Thank you check and fix my code. I will be careful in view! macro from next time.

**Pull request merged**